frontend/pages/queries/edit/components/DiscardDataOption/DiscardDataOption.tsx

```diff
@@ -0,0 +1,101 @@
+import Checkbox from "components/forms/fields/Checkbox";
+import Icon from "components/Icon";
+import InfoBanner from "components/InfoBanner";
+import TooltipWrapper from "components/TooltipWrapper";
+import { IConfig } from "interfaces/config";
+import { QueryLoggingOption } from "interfaces/schedulable_query";
+import React, { useState } from "react";
+import { Link } from "react-router";
+
+const baseClass = "discard-data-option";
+
+interface IDiscardDataOption {
+  appConfig?: IConfig;
+  selectedLoggingType: QueryLoggingOption;
+  discardData: boolean;
+  setDiscardData: (value: boolean) => void;
+  breakHelpText?: boolean;
+}
+
+const DiscardDataOption = ({
+  appConfig,
+  selectedLoggingType,
+  discardData,
+  setDiscardData,
+  breakHelpText = false,
+}: IDiscardDataOption) => {
+  const [forceEditDiscardData, setForceEditDiscardData] = useState(false);
+  const disable =
+    appConfig?.server_settings.query_reports_disabled && !forceEditDiscardData;
+
+  return (
+    <div className={baseClass}>
+      {["differential", "differential_ignore_removals"].includes(
+        selectedLoggingType
+      ) && (
+        <InfoBanner color="purple-bold-border">
+          <>
+            The <b>Discard data</b> setting is ignored when differential logging
+            is enabled. This <br />
+            query&apos;s results will not be saved in Fleet.
+          </>
+        </InfoBanner>
+      )}
+      <Checkbox
+        name="discardData"
+        onChange={setDiscardData}
+        value={discardData}
+        wrapperClassName={
+          disable ? `${baseClass}__disabled-discard-data-checkbox` : ""
+        }
```

Comment:
Adding optional `disableLabel` prop for now – @rachaelshaw agrees with the idea of making this behavior global for all checkboxes, but has to go through feature fest: [slack](https://fleetdm.slack.com/archives/C01EZVBHFHU/p1696614843490569?thread_ts=1696612917.643429&cid=C01EZVBHFHU)

Reply:
Actually, going to leave this as-is, since in this case we are being asked to allow the disabled checkbox to be selected, but per above slack comment, tbd about globally with checkboxes.

Reply:
Added to feature fest for consideration: https://docs.google.com/document/d/1mwu5WfdWBWwJ2C3zFDOMSUC9QCyYuKP4LssO_sIHDd0/edit, #17